tests/notifications.rs

```diff
@@ -0,0 +1,72 @@
+pub mod support;
+
+use std::time::Duration;
+
+use crate::support::registry::create_registry;
+use stand_control_bot::logic::notifications::{Notification, NotificationsRecevier};
+use tokio::sync::mpsc::Sender;
+use tokio::{sync::mpsc, time::timeout};
+
+use anyhow::{Context, Result};
+use stated_dialogues::{controller::BotAdapter, dialogues::MessageId};
+
+struct TestAdapter {
+    sender: Sender<String>,
```

Comment:
There is no need to use channels here, sent messages can just be stored in pub `Vec` or `HashMap` to access them from tests

Reply:
Then method send_message must be mutable, but we use trait and can't override it

Reply:
Not really, there is always [RefCell](https://doc.rust-lang.org/std/cell/struct.RefCell.html)